client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -192,8 +192,8 @@ function mediaButton( editor ) {
 
 				// When merging, allow any updated field to be used if it doesn't
 				// already exist in the current markup, but otherwise only force
-				// update ID and URL attributes to their new values
-				const merged = assign( {}, media, current.media, pick( media, 'ID', 'URL' ), {
+				// update ID, URL, width and height attributes to their new values
+				const merged = assign( {}, media, current.media, pick( media, 'ID', 'URL', 'width', 'height' ), {
```

Comment:
Actually I found another case:
1. Throttle your connection to something slow
2. Drag and drop a large image directly into the post content
3. Resize to something much smaller
4. Set network connection back to no throttle
5. We update the image back to full size